lib/conncache.c

```diff
@@ -395,7 +395,6 @@ bool Curl_conncache_return_conn(struct Curl_easy *data,
          important that details from this (unrelated) disconnect does not
          taint meta-data in the data handle. */
       struct conncache *connc = data->state.conn_cache;
-      connc->closure_handle->state.buffer = data->state.buffer;
       connc->closure_handle->set.buffer_size = data->set.buffer_size;
```

Comment:
Maybe `closure->buffer_size` should not be assigned at all here, as then I believe it will remain at zero which should be a better value to detect the error if this is actually used when shutting down?

Reply:
I can do that and error the borrow if the transfer has size 0?

Reply:
Changed as requested, transfer with buffer_size 0 trying to borrow errors now.